### PR TITLE
add(resolver): Allow async before and after

### PR DESCRIPTION
**Pull request by janmeier**

### CODE REVIEW

File: src/resolver.js [hunk 1]
```diff
@@ -103,12 +103,12 @@ function resolverFactory(target, options) {
     findOptions.context = context;
     findOptions.logging = findOptions.logging || context.logging;
 
-    findOptions = options.before(findOptions, args, context, {
+    findOptions = await Promise.resolve(options.before(findOptions, args, context, {
```
Comment (mickhansen) [inline src/resolver.js]: Not using async/await in codebase

Comment (mickhansen) [inline src/resolver.js]: Should also be applied here: https://github.com/mickhansen/graphql-sequelize/blob/master/src/generateIncludes.js#L75

File: test/integration/resolver.test.js [hunk 2]
```diff
@@ -1039,6 +1039,59 @@ describe('resolver', function () {
...
+                type: GraphQLInt
+              },
+              order: {
+                type: GraphQLString
+              }
+            },
+            resolve: resolver(User, {
+              before: async function (options) {
+                await Promise.delay(100);
+                return options;
```
Comment (mickhansen) [inline test/integration/resolver.test.js]: returning a promise should be sufficient to test it

Comment (janmeier) [inline test/integration/resolver.test.js]: The resolver expects before to return options

Comment (mickhansen) [inline test/integration/resolver.test.js]: Right, `return Promise.resolve(options)`

**Comment by janmeier** [comment]
Updated - I recommend viewing with `?w=1` since there are a lot of whitespace changes :)

File: src/generateIncludes.js [hunk 3]
```diff
@@ -63,67 +61,70 @@ export default function generateIncludes(simpleAST, type, context, options) {
...
-      }
+      return new Promise(function (resolve) {
+        if (includeResolver.$before) {
+          resolve(includeResolver.$before(includeOptions, args, context, {
+            ast: fieldAST,
+            type: type
+          }));
+        } else {
+          resolve(includeOptions);
+        }
```
Comment (mickhansen) [inline src/generateIncludes.js]: Why new promise? If it's because you need a chain `Promise.resolve().then(` might be better.

Comment (janmeier) [inline src/generateIncludes.js]: Updated (:

**Comment by janmeier** [comment]
Linted - and added a precommit hook so it doesn't happen again ;)

**Comment by mickhansen** [comment]
1.2.0

**Pull request merged**